core/trino-main/src/main/java/io/trino/server/security/Authenticator.java

```diff
@@ -13,12 +13,50 @@
  */
 package io.trino.server.security;
 
+import io.trino.client.ProtocolDetectionException;
 import io.trino.spi.security.Identity;
 
 import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MultivaluedMap;
+
+import java.security.Principal;
+import java.util.Optional;
+
+import static io.trino.client.ProtocolHeaders.detectProtocol;
 
 public interface Authenticator
 {
     Identity authenticate(ContainerRequestContext request)
             throws AuthenticationException;
+
+    default void checkAndRewriteUserHeaderToMappedUserWhenRequired(
```

Comment:
This method should not be part of the interface. Also interfaces should not be used for as utility classes.

Reply:
@kokosing any suggestion to which utility i should move it ? Or create a new one ?

Reply:
Please create a new one.

Reply:
Applied